core/src/main/java/io/grpc/internal/DelayedStream.java

```diff
@@ -109,6 +110,7 @@ public void appendTimeoutInsight(InsightBuilder insight) {
       } else {
         insight.appendKeyValue("buffered_nanos", System.nanoTime() - startTimeNanos);
         insight.append("waiting_for_connection");
+        insight.append(lastPickStatus);
```

Comment:
There's no context here. It won't be clear what this means. Add a key to describe it.